odoo/addons/base/models/ir_http.py

```diff
@@ -487,3 +489,24 @@ def _response_by_status(self, status, headers, content):
             return werkzeug.utils.redirect(content, code=301)
         elif status != 200:
             return request.not_found()
+
+    @classmethod
+    def _get_session_store(cls):
+        path = odoo.tools.config.session_dir
+        _logger.debug('HTTP sessions stored in: %s', path)
+        return sessions.FilesystemSessionStore(
+            path, session_class=OpenERPSession, renew_missing=True)
+
+    @api.autovacuum
+    def _session_vacuum(self):
+        # we keep session one week
+        session_store = self._get_session_store()
+
+        last_week = time.time() - 60*60*24*7
```

Comment:
```python
number_day = int(self.env['ir.config_parameter'].sudo().get_param("http.session_duration", 7))
last_week = time.time() - 60*60*24*number_day
```
in some e-commerce website, it is nice to keep session more longer than 7 days.

@odony @rco-odoo @JKE-be

Reply:
Right, this could be a config parameter that default to 7 days.

Reply:
I rewrote the function using Pathlib. Check this out